### PR TITLE
Release/test release

### PR DESCRIPTION
**Pull request by maxschulz-COL**
## Description
- implement `hatch` commands that allow for release prepping and execution

The workflow for manual releasing would be as follows:

0. Create a branch with an appropriate name: `release/vX.Y.Z
1. Run `hatch run prep-release` - check all files created, commit and raise and merge the PR
2. Checkout the merge commit on `main` that preps the release
3. Run `hatch run release` - one can add the options `-r main` or `-r test` to publish to PyPI or testPyPI (tokens are needed to authenticate)

## Checklist

- [x] I have not referenced individuals, products or companies in any commits, directly or indirectly
- [x] I have not added data or restricted code in any commits, directly or indirectly
- [x] I have updated the docstring of any public function/class/model changed
- [ ] I have added the PR number to the change description in the changelog fragment, e.g. `Enable feature XXX (#123)` (if applicable)

## Types of changes

- [x] Docs/refactoring (non-breaking change which improves codebase)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

### CODE REVIEW

File: vizro-core/hatch.toml [hunk 1]
```diff
@@ -37,7 +37,20 @@ cov-report = [
...
+  "echo 'Raise a PR to prepare main for release'"
+]
 pypath = "hatch run python -c 'import sys; print(sys.executable)'"
+release = [
+  "git tag -a v`hatch version` -m 'Release version'",
+  "git push origin v`hatch version`",
+  "hatch build",
+  "hatch publish {args}",
+  "hatch version patch,dev",
+  "echo 'Raise a PR to bump to the new development version'"
```
Comment (huong-li-nguyen) [inline vizro-core/hatch.toml:52]: Is there a way we can add the version number in the string?

```suggestion
  "echo 'Raise a PR to bump to the new development version'"
```

**Review by huong-li-nguyen — CHANGES_REQUESTED** [review]
Can you add the description of the manual process in the dev docs?

**Review by petar-qb — APPROVED** [review]
For me this looks good!

**Comment by maxschulz-COL** [comment]
> Will approve now to not block you, but would be great to add the steps somewhere. I don't mind to have it in the contributing guide. It should just be easy to find :)

Will find a place for this along some other improvements in another PR :)

**Pull request merged**